### PR TITLE
Update TLS support

### PR DESCRIPTION
**Pull request by networkfusion**
## Description

TLS 1.0 and 1.1 have been removed as no longer supported.
TLS 1.3 has been added.

## Motivation and Context

Keeps alignment with latest firmware TLS release nanoframework/nf-interpreter#2989.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: M2Mqtt/Net/MqttNetworkChannel.cs [hunk 1]
```diff
@@ -320,14 +320,10 @@ public static SslProtocols ToSslPlatformEnum(MqttSslProtocols mqttSslProtocol)
             {
                 case MqttSslProtocols.None:
                     return SslProtocols.None;
-                case MqttSslProtocols.TLSv1_0:
```
Comment (Ellerbach) [inline M2Mqtt/Net/MqttNetworkChannel.cs:323]: why removing them? they are still valid.

Comment (networkfusion) [reply M2Mqtt/Net/MqttNetworkChannel.cs:323]: https://github.com/nanoframework/nf-interpreter/pull/2989

mainly: 
image

Comment (networkfusion) [reply M2Mqtt/Net/MqttNetworkChannel.cs:323]: I think @josesimoes  forgot to do the announcement...

**Review by josesimoes — APPROVED** [review]
LGTM! Thanks

**Pull request merged**